File: frontend/packages/core/src/validations/is-email/is-email.ts
const isEmail = (input: string): boolean => {
  const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
  const domainPart = input.split('@')[1];
  if (domainPart?.includes('..')) {
    return false;
  }

  return emailRegex.test(input);
};

export default isEmail;

File: mobile/packages/footprint-swift/Sources/FootprintSwift/OnboardingComponents/Utils/Validations/IsEmail.swift
import Foundation

func isEmail(_ input: String) -> String? {
    if input.isEmpty {
        return "Email is required"
    }
    
    let emailRegex = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$"
    let domainParts = input.split(separator: "@")
    
    if domainParts.count > 1, domainParts[1].contains("..") {
        return "Invalid email format"
    }
    
    let isValidEmail = NSPredicate(format: "SELF MATCHES %@", emailRegex).evaluate(with: input)
    if !isValidEmail {
        return "Invalid email format"
    }
    
    return nil
}
